src/timeunit.ts

```diff
@@ -52,6 +52,54 @@ export function isSingleTimeUnit(timeUnit: TimeUnit) {
   return !!SINGLE_TIMEUNIT_INDEX[timeUnit];
 }
 
+/**
+ * Converts a date to only have the measurements relevant to the specified unit
+ * i.e. ('yearmonth', '2000-12-04 07:58:14') -> '2000-12-00 00:00:00'
+ */
+export function convert(unit: TimeUnit, date: Date, utc: boolean): Date {
+  const result: Date = new Date(0, 0, 0, 0, 0, 0, 0); // start with uniform date
+  // handle some units individually
+  if (unit === TimeUnit.QUARTERMONTH) {
+    // not sure how to handle this unit. Just put month (since you can infer quarter from month)?
+  } else {
+    SINGLE_TIMEUNITS.forEach(function(singleUnit) {
+      if (containsTimeUnit(unit, singleUnit)) {
+        // can ignore TimeUnit.DAY
```

Comment:
add a case on top to throw error if day is input?  (We will just not support convert)